src/ray/rpc/grpc_client.h

```diff
@@ -90,12 +83,12 @@ class GrpcClient {
   ///
   /// \return Status.
   template <class Request, class Reply>
-  ray::Status CallMethod(
+  void CallMethod(
       const PrepareAsyncFunction<GrpcService, Request, Reply> prepare_async_function,
       const Request &request, const ClientCallback<Reply> &callback) {
     auto call = client_call_manager_.CreateCall<GrpcService, Request, Reply>(
         *stub_, prepare_async_function, request, callback);
-    return call->GetStatus();
+    RAY_CHECK(call != nullptr);
```

Comment:
@zhijunfu Doesn't this throw away bad statuses? I was expecting something more like:

```
auto send = call->GetStatus();
if (!send.ok()) {
   callback(send, Reply());
}
```

Reply:
@ericl The assumption here is that the callback will be invoked by Gprc, no matter it succeeds or fails. With this assumption, the callback will be invoked twice if we call it here on failure as well.

Reply:
@stephanie-wang is this true?